components/private/Math/src/MDCMath.h

```diff
@@ -125,3 +125,35 @@ static inline CGRect MDCRectAlignToScale(CGRect rect, CGFloat scale) {
                     MDCCeil((CGRectGetWidth(rect) + adjustWidthHeight.width) * scale) / scale,
                     MDCCeil((CGRectGetHeight(rect) + adjustWidthHeight.height) * scale) / scale);
 }
+
+static inline CGPoint MDCPointRoundWithScale(CGPoint point, CGFloat scale) {
+  if (MDCCGFloatEqual(scale, 0)) {
+    return CGPointZero;
+  }
+
+  return CGPointMake(MDCRound(point.x * scale) / scale, MDCRound(point.y * scale) / scale);
+}
+
+/**
+ Align the centerPoint of a view so that its origin is pixel-aligned. Returns @c CGRectZero if
+ @c scale is zero or @c bounds is @c CGRectNull.
+
+ @param center the unaligned center of the view.
+ @param bounds the bounds of the view.
+ @param scale the native scaling factor for pixel alignment.
+
+ @return the center point of the view such that its origin will be pixel-aligned.
+ */
+static inline CGPoint MDCRoundCenterWithBoundsAndScale(CGPoint center,
+                                                       CGRect bounds,
+                                                       CGFloat scale) {
+  if (MDCCGFloatEqual(scale, 0) || CGRectIsNull(bounds)) {
```

Comment:
|| CGRectIsEmpty(bounds) ?

Reply:
If the rect is empty, that implies a zero height and width - should we not still align the point?

Reply:
Would you prefer that check to just return `MDCPointRoundWithScale()`?

Reply:
Ah, never mind. Although a CGRect is empty, it may still have a height or width (only one needs to be 0). In that case, the math to adjust the center point still needs to take place.

Reply:
Please disregard, we definitely want to align the point in case of an empty rect.